perceval/backends/core/github.py

```diff
@@ -120,83 +147,112 @@ def fetch(self, from_date=DEFAULT_DATETIME):
 
         from_date = datetime_to_utc(from_date)
 
-        issues_groups = self.client.get_issues(from_date)
+        issues_groups = self.client.get_paginated_items("/issues", payload_type="issue", start=from_date)
 
         for raw_issues in issues_groups:
+            self._push_cache_queue('{ISSUES}')
             self._push_cache_queue(raw_issues)
             self._flush_cache_queue()
             issues = json.loads(raw_issues)
             for issue in issues:
-                for field in ['user', 'assignee']:
+                for field in TARGET_ISSUE_FIELDS:
+                    issue[field + "_data"] = {}
+
                     if issue[field]:
-                        issue[field + "_data"] = self.__get_user(issue[field]['login'])
-                    else:
-                        issue[field + "_data"] = {}
+                        if field == "user":
+                            issue[field + "_data"] = self.__get_user(issue[field]['login'])
+                        elif field == "assignees":
+                            issue[field + "_data"] = self.__get_issue_assignees(issue[field])
+                        elif field == "comments":
+                            issue[field + "_data"] = self.__get_issue_comments(issue["number"])
+                self._push_cache_queue('{}')
+                self._flush_cache_queue()
                 yield issue
 
     @metadata
     def fetch_from_cache(self):
         """Fetch the issues from the cache.
-
         It returns the issues stored in the cache object provided during
         the initialization of the object. If this method is called but
         no cache object was provided, the method will raise a `CacheError`
         exception.
-
         :returns: a generator of items
-
         :raises CacheError: raised when an error occurs accessing the
             cache
         """
         if not self.cache:
             raise CacheError(cause="cache instance was not provided")
 
         cache_items = self.cache.retrieve()
-
+        current_issue = None
         issues = None
 
         while True:
             try:
                 raw_item = next(cache_items)
+
```

Comment:
I think you should split this work in several and small methods. It would be easy to read and to understand. One method for processing users, one for assignees and so on.